pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -1955,10 +1955,11 @@ public boolean getBoolean(int columnIndex) throws SQLException {
     }
 
     if (isBinary(columnIndex)) {
-      return BooleanTypeUtil.castToBoolean(readDoubleValue(this_row[col], fields[col].getOID(), "boolean"));
+      return BooleanTypeUtil.castToBoolean(readDoubleValue(this_row[col], fields[col].getOID(), "boolean"),
+          Oid.toString(fields[col].getOID()));
```

Comment:
It's in an error handler, this does not concern me much, although we could cache it in the event that someone had *alot* of errors